### ISSUE TITLE
Safe Exam Browser does not detect any displays.

### ISSUE THREAD

**Issue by FranciscoRibeiro03** [SafeExamBrowser/seb-win-refactoring#268]
**Describe the Bug**
I had to take a test on my University's elearning platform (Moodle), and when I tried to start SEB, it showed an error message saying that "0 internal and 0 external displays were detected" (screenshot below)

**Steps to Reproduce**
1. Install SEB (started on version 3.2.0, I tried version 3.1.1 and it worked)
2. Try to take a test where the config locks the amount of displays to either 1, 2 or 3.
3. See error

**Expected Behavior**
SEB should detect the display and start the test with no errors.

**Screenshots**
![image](https://user-images.githubusercontent.com/15705566/143899272-51e45df0-2f92-4198-b8c8-bea609a5f2d7.png)

**Version Information**
 - OS: Windows 10 Professional, Version 21H2 (OS Build 19044.1348)
 - SEB Version 3.3.1.388

**Additional Context**
I am using a laptop (ASUS ROG Strix GL703VD) with an i7-7700HQ, a GTX 1050 and 16GB of RAM, if that is of any interest.
I also found an error on the log, found on `%LOCALAPPDATA%\SafeExamBrowser\Logs`. I'm attaching the log, in case it helps.
[2021-11-29_15h49m02s_Runtime.log](https://github.com/SafeExamBrowser/seb-win-refactoring/files/7619354/2021-11-29_15h49m02s_Runtime.log)

**Comment by dbuechel** [comment]
Could you please try to reset the WMI repository: https://techcommunity.microsoft.com/t5/ask-the-performance-team/wmi-rebuilding-the-wmi-repository/ba-p/373846

**Comment by jeroenhabets** [comment]
@dbuechel I encountered the same issue while running the beta SEB for Windows in a Virtual Box VM (latest VB).
FYI: In an Administrator CMD prompt both `winmgmt /verifyrepository` and `winmgmt /salvagerepository` returned "WMI repository is consistent"

**Comment by dbuechel** [comment]
Yes, in VMs the check fails for some reason. Hence we added the option to ignore validation errors in the configuration.

**Comment by RishabhParwal** [comment]
@dbuechel I am also getting the same error
![image](https://user-images.githubusercontent.com/67789504/146227765-f27b2ee1-31b7-436d-aa3c-cab97deb7a5e.png)

**Comment by dbuechel** [comment]
Thanks for the update, @rui2015. Glad to know you were able to make it work, albeit with having to reset your entire machine. Yes, it must be some configuration or system component which can get corrupted. Most unfortunately, we had no luck thus far in pinning down the root cause.

**Comment by dbuechel** [comment]
We were able to gain more information, it namely appears to be the case that this error can be caused by a specific display driver from Windows ("PnP-Monitor Lock (Standard)"), see #341.

**Comment by FrassettoN** [comment]
Hello,
I'm facing the same problem 

# Logfile: 
```
/* Safe Exam Browser, Version 3.3.2 (x64), Build 3.3.2.413
/* Copyright © 2022 ETH Zürich, Educational Development and Technology (LET)
/* 
/* Please visit https://www.github.com/SafeExamBrowser for more information.

# Application started at 2022-04-10 16:28:27.264
# Running on Windows 10, Microsoft Windows NT 10.0.22593.0 (x64)
# Computer 'DESKTOP-3S74I5P' is a XPS XPS 13 7390 manufactured by Dell Inc.
# Runtime-ID: 5aad2db8-68af-4244-a8aa-0122948b40a6

2022-04-10 16:28:27.736 [04] - INFO: Initiating startup procedure...
2022-04-10 16:28:27.751 [04] - INFO: Loading text data...
2022-04-10 16:28:27.765 [04] - DEBUG: [Text] System language is 'it', trying to load data...
2022-04-10 16:28:27.765 [04] - DEBUG: [Text] Data successfully loaded.
2022-04-10 16:28:27.765 [04] - INFO: Starting communication host...
2022-04-10 16:28:27.813 [09] - DEBUG: [RuntimeHost] Communication host is opening...
2022-04-10 16:28:27.863 [09] - DEBUG: [RuntimeHost] Communication host has been opened.
2022-04-10 16:28:27.863 [09] - DEBUG: [RuntimeHost] Successfully started communication host for endpoint 'net.pipe://localhost/safeexambrowser/runtime/582ba169-006c-463b-9900-69def67a0321'.
2022-04-10 16:28:27.874 [04] - INFO: Application successfully initialized.

2022-04-10 16:28:27.936 [04] - INFO: ### ------------------------------------ Session Start Procedure ------------------------------------- ###
2022-04-10 16:28:27.944 [04] - INFO: Initializing new session configuration...
2022-04-10 16:28:27.946 [04] - INFO:  -> Client-ID: 0c149f21-506e-4017-8c35-da01ea8a3262
2022-04-10 16:28:27.946 [04] - INFO:  -> Runtime-ID: 5aad2db8-68af-4244-a8aa-0122948b40a6
2022-04-10 16:28:27.949 [04] - INFO:  -> Session-ID: c9470eb9-66db-48b7-bd5d-05f2da91cc2c
2022-04-10 16:28:27.953 [04] - INFO: Initializing application configuration...
2022-04-10 16:28:27.955 [04] - INFO: No valid configuration resource specified and no local client configuration found - loading default settings...
2022-04-10 16:28:27.971 [04] - INFO: The configuration was successful.
2022-04-10 16:28:27.971 [04] - INFO: Remote proctoring is disabled, skipping disclaimer.
2022-04-10 16:28:27.971 [04] - INFO: Validating remote session policy...
2022-04-10 16:28:27.983 [04] - DEBUG: [RemoteSessionDetector] System appears not to be running in a remote session.
2022-04-10 16:28:27.983 [04] - INFO: Validating virtual machine policy...
2022-04-10 16:28:27.986 [04] - DEBUG: [VirtualMachineDetector] Computer 'DESKTOP-3S74I5P' appears to not be a virtual machine.
2022-04-10 16:28:27.986 [04] - INFO: Validating display configuration...
2022-04-10 16:28:28.067 [04] - ERROR: [DisplayMonitor] Failed to query displays!

   Exception Message: Accesso negato 
   Exception Type: System.Management.ManagementException

   in System.Management.ManagementException.ThrowWithExtendedInfo(ManagementStatus errorCode)
   in System.Management.ManagementObjectCollection.ManagementObjectEnumerator.MoveNext()
   in System.Linq.Enumerable.<CastIterator>d__97`1.MoveNext()
   in SafeExamBrowser.Monitoring.Display.DisplayMonitor.TryLoadDisplays(IList`1& displays) in C:\Users\appveyor\projects\seb-win-refactoring\SafeExamBrowser.Monitoring\Display\DisplayMonitor.cs:riga 169

2022-04-10 16:28:28.071 [04] - WARNING: [DisplayMonitor] Failed to validate display configuration, active configuration is not allowed.
2022-04-10 16:28:28.073 [04] - ERROR: Display configuration is not allowed!
2022-04-10 16:28:31.463 [04] - INFO: ### -------------------------------------- Session Start Failed -------------------------------------- ###

2022-04-10 16:28:32.455 [07] - INFO: Initiating shutdown procedure...
2022-04-10 16:28:32.460 [07] - INFO: Stopping communication host...
2022-04-10 16:28:32.465 [07] - DEBUG: [RuntimeHost] Communication host is closing...
2022-04-10 16:28:32.501 [07] - DEBUG: [RuntimeHost] Communication host has been closed.
2022-04-10 16:28:32.503 [07] - DEBUG: [RuntimeHost] Terminated communication host for endpoint 'net.pipe://localhost/safeexambrowser/runtime/582ba169-006c-463b-9900-69def67a0321'.
2022-04-10 16:28:32.505 [07] - INFO: Application successfully finalized.

# Application terminated at 2022-04-10 16:28:32.555
```

# What I've done so far:
- reinstalled the application 
- searched for the driver "PnP-Monitor Lock (Standard)" (under devices -> DESKTOP-3S74I5P -> monitor) without results
-  ran in an Administrator CMD prompt `winmgmt /verifyrepository` and `winmgmt /salvagerepository`, but both returned "WMI repository is consistent"

# Note
I'm currently running Windows 11 Pro with Windows Subsystem for Linux enabled. 

@dbuechel any new idea?
Thank you so much

**Comment by dbuechel** [comment]
Thanks for the information, @FrassettoN. We have a testing device where the issue apparently can be reproduced, I hope that I can have a look at it in the coming weeks. Until now, the assumption was that the issue has something to do with the display driver mentioned above, but if that isn't the case on your machine, then we're basically back to zero...

**Comment by dbuechel** [comment]
> I have installed version 3.1.1.250 (from CISIA website) and it works perfectly.

That's because the display monitoring was a new feature which we introduced in version 3.2.0 😉

**Comment by FrassettoN** [comment]
> That's because the display monitoring was a new feature which we introduced in version 3.2.0 😉

Yeah, yeah, I know. It was more a tip for those who will encounter this problem a few days before a test 😂

**Comment by jeroenhabets** [comment]
Hi @dbuechel, are you keeping this open because of the VM display detection issue? Though, it would be nice to get fixed from my personal perspective, I seriously doubt it's hampering many (real) exam takers... Regards, Jeroen

**Comment by dbuechel** [comment]
@jeroenhabets Yes exactly. I'll attempt to fix the problem with the display driver mentioned above once I have time to look at our testing device, but it may well be that we'll have to classify this as "won't fix" for the moment. I thus far was unfortunately not able to reproduce the issue.

**Comment by dbuechel** [comment]
I have to mark this as won't fix for the moment. I was unable to get to the root cause of the issue (I still suspect some issue in the .NET Framework / WMI, possibly related to the display driver mentioned above) and also failed to find a solution or workaround to the problem. I furthermore can only reproduce the issue on one particular device, and only when I keep the laptop lid shut directly after starting the machine.

**Comment by eliazarx** [comment]
Hello everyone,
I have the same problem when i use the original driver AMD Radeon(TM) Vega 11 Graphics.

[2022-08-12_13h14m45s_Runtime.log](https://github.com/SafeExamBrowser/seb-win-refactoring/files/9315898/2022-08-12_13h14m45s_Runtime.log)

When I use the generic microsoft driver I no longer have the problem and SEB starts correctly.

**Comment by dbuechel** [comment]
According to the log, you do not have the same issue:

```
2022-08-12 13:14:55.792 [06] - INFO: Validating display configuration...
2022-08-12 13:14:55.807 [06] - INFO: [DisplayMonitor] Detected active, external display 'DISPLAY\LEN62B6\5&3bac39b&0&UID268_0' connected via 'HDMI'.
2022-08-12 13:14:55.807 [06] - INFO: [DisplayMonitor] Detected 1 active displays, 3 are allowed.
2022-08-12 13:14:55.807 [06] - WARNING: [DisplayMonitor] Detected external display but only internal displays are allowed!
2022-08-12 13:14:55.807 [06] - ERROR: Display configuration is not allowed!
```

See the second to last line: **Detected external display but only internal displays are allowed!**

**Comment by eliazarx** [comment]
You're right!

Should I open a new thread?

**Comment by dbuechel** [comment]
It appears that you're using an external display, but the configuration allows only internal displays. Is that correct?

**Comment by eliazarx** [comment]
No. I'm using a normal desktop pc with a single monitor.

The problem is that when I use the official video card driver I encounter this problem. If I use a generic microsoft driver I have no problem.

**Comment by eliazarx** [comment]
What I don't explain is: why don't I have problems if I use the generic microsoft driver?

**Comment by dbuechel** [comment]
> Why don't I have problems if I use the generic microsoft driver?

That indeed is a good question. Could you provide some screen shots / photos and other details of the exact setup that you're using?

**Comment by hunterz-killer** [comment]
@dbuechel hi

I'm having a similar problem 

Actually I'm using Mettl's MSB which in backend SEB.

I used the same PC for last semester exam and it worked well. But now my college instructed my to update the software (Mettl MSB) after which I getting **0 internal and 0 external displays detected** error.

I have attached the screenshot and log for your reference.

Note: My Graphic Driver is Intel UHD 550 and monitor Driver is Generic Non-PnP Monitor.

![IMG_20221022_195400_897](https://user-images.githubusercontent.com/82221655/197401858-c3808f94-7568-4c61-8ab7-461e69f8190e.jpg)

![IMG_20221022_195420_096](https://user-images.githubusercontent.com/82221655/197401865-65cf7491-54c8-4e2e-ba96-8b7d979d554b.jpg)


[2022-10-22_19h41m46s_Runtime.log.txt](https://github.com/SafeExamBrowser/seb-win-refactoring/files/9846968/2022-10-22_19h41m46s_Runtime.log.txt)

Thanks in advance

**Comment by dbuechel** [comment]
@hunterz-killer We cannot help you with MSB, that's another product and you'd need to contact its manufacturers if you have issues with it. According to the screenshots above, you're trying to use a custom build, is that correct?

**Comment by hunterz-killer** [comment]
> @hunterz-killer We cannot help you with MSB, that's another product and you'd need to contact its manufacturers if you have issues with it. According to the screenshots above, you're trying to use a custom build, is that correct?

Yes, you are correct it seems to a custom build but i couldn't confirm it.

I have reached the MSB support team but couldn't solve it.
Is there any possibility or things I can check with to solve the issue.

**Comment by dbuechel** [comment]
Apart from the driver issue mentioned above, we unfortunately do not have any indications of what the underlying issue could be. We didn't encounter the problem ourselves thus far (apart from the special case above), so there's really not too much we can do until we get a reliable way to reproduce the issue meaningfully.

**Comment by KingArnim** [comment]
So, Guys who are facing the above mentioned error please follow the steps below:-

1) open this link -> https://www.entechtaiwan.com/util/moninfo.shtm

2) download the software

3) install it and open it

4) go on top left click on file option and click on create Imf file and save it 

5) once done remember the location you saved that file on.

6) now go to system settings,then go to update and security and go to recovery and click restart now

7) when you are in recovery click on troubleshoot and select advanced options and select startup settings

8) on the new window opened (if it ask you to select account then select your account) after that press 7 on your keyboard and your
computer may restart

9) now go to device manager, then drop down the monitor tab, then right click on the "generic non-pnp monitor" and click update driver

10) now select "browse my computer drivers" and then click "let me pick" then click on have disk.

11) now click on "browse" and then select the imf file we saved in starting and once it is installed let the device manager refresh itself.

12) now again right click on what so ever name is there in place of "generic non-pnp monitor" and go to update driver but this time click on select driver automatically

13) once it is done just run msb again and check the error must get solved ❤️✌️

Solved by @hunterz-killer after a long communication with him 😃👍

**Comment by dajahaniquek** [comment]
Im having the same issue on my laptop
